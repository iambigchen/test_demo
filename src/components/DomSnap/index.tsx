import { FC } from "react";

interface IProps {}

// 12 | 快照测试：怎么保障组件 UI 的完整？
export const DomSnap: FC<IProps> = ({}) => {
  return (
    <form aria-label="form">
      <input
        type="text"
        name="username"
        disabled
        aria-disabled
        defaultValue="zhenmin"
        aria-label="form_username"
      />
      <input
        type="number"
        name="age"
        defaultValue={23}
        required
        aria-label="form_age"
      />
      <input
        type="radio"
        name="sex"
        value="man"
        defaultChecked
        aria-checked
        aria-label="form_sex"
      />
      <input type="radio" name="sex" value="woman" aria-label="form_sex" disabled/>
    </form>
  );
};