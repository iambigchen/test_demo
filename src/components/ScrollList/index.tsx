import { FC, useMemo, useState } from "react";
import "./styles.css";

interface IProps {
  data: string[];
  height: string | number;
  pageSize?: number;
}

/**
 * 滚动 list, 拉到底部刷新新的一页
 * @param data
 * @param height
 * @returns
 */
export const ScrollList: FC<IProps> = ({ data, height, pageSize = 10 }) => {
  const [page, setPage] = useState(1);

  const currentData = useMemo(
    () => data.slice(0, pageSize * page),
    [pageSize, page]
  );

  return (
    <div
      className="scrollList"
      style={{ height }}
      onScroll={(e) => {
        const { scrollTop, clientHeight, scrollHeight } = e.currentTarget;
        if (
          scrollTop + clientHeight >= scrollHeight &&
          currentData.length < data.length
        ) {
          alert(`当前page为${page}`);
          setPage(page + 1);
        }
      }}
    >
      {currentData.map((item, index) => {
        return (
          <div className="item" key={index}>
            {item}
          </div>
        );
      })}
    </div>
  );
};
