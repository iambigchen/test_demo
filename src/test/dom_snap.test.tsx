import React from "react";
import { render, screen } from "@testing-library/react";
import { DomSnap } from "../components/DomSnap";

// 12 | 快照测试：怎么保障组件 UI 的完整？
describe("examples for snap", () => {
  test("a test for component snap", () => {
    const { baseElement } = render(<DomSnap />);
    expect(baseElement).toMatchSnapshot();
  });

  test("a test for part component snap", () => {
    render(<DomSnap />);
    expect(
      screen.getByRole("textbox", { name: "form_username" })
    ).toMatchSnapshot();
  });
});