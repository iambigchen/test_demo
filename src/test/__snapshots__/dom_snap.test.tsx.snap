// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`examples for snap a test for component snap 1`] = `
<body>
  <div>
    <form
      aria-label="form"
    >
      <input
        aria-disabled="true"
        aria-label="form_username"
        disabled=""
        name="username"
        type="text"
        value="zhenmin"
      />
      <input
        aria-label="form_age"
        name="age"
        required=""
        type="number"
        value="23"
      />
      <input
        aria-checked="true"
        aria-label="form_sex"
        checked=""
        name="sex"
        type="radio"
        value="man"
      />
      <input
        aria-label="form_sex"
        disabled=""
        name="sex"
        type="radio"
        value="woman"
      />
    </form>
  </div>
</body>
`;

exports[`examples for snap a test for part component snap 1`] = `
<input
  aria-disabled="true"
  aria-label="form_username"
  disabled=""
  name="username"
  type="text"
  value="zhenmin"
/>
`;
